src/BuildCheck.UnitTests/EndToEndTests.cs

```diff
@@ -27,6 +27,8 @@ public EndToEndTests(ITestOutputHelper output)
         _env.WithEnvironmentInvariant();
     }
 
+    private static string TestAssetsRootPath { get; } = Path.Combine(AppContext.BaseDirectory, "TestAssets");
```

Comment:
Is this going to be correct for all execution scenarios of tests? Should we use [`Assembly.Location`](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.location?view=net-8.0) instead?